deeplearning4j-core/src/test/java/org/deeplearning4j/gradientcheck/CNN3DGradientCheckTest.java

```diff
@@ -0,0 +1,113 @@
+package org.deeplearning4j.gradientcheck;
+
+import org.deeplearning4j.BaseDL4JTest;
+import org.deeplearning4j.nn.conf.ConvolutionMode;
+import org.deeplearning4j.nn.conf.MultiLayerConfiguration;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.distribution.NormalDistribution;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.conf.layers.*;
+import org.deeplearning4j.nn.conf.preprocessor.Cnn3DToFeedForwardPreProcessor;
+import org.deeplearning4j.nn.conf.preprocessor.RnnToFeedForwardPreProcessor;
+import org.deeplearning4j.nn.layers.convolution.Convolution3DLayer;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.junit.Test;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.api.buffer.DataBuffer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.learning.config.NoOp;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CNN3DGradientCheckTest extends BaseDL4JTest {
+    private static final boolean PRINT_RESULTS = true;
+    private static final boolean RETURN_ON_FIRST_FAILURE = false;
+    private static final double DEFAULT_EPS = 1e-6;
+    private static final double DEFAULT_MAX_REL_ERROR = 1e-3;
+    private static final double DEFAULT_MIN_ABS_ERROR = 1e-8;
+
+    static {
+        Nd4j.setDataType(DataBuffer.Type.DOUBLE);
+    }
+
+    @Test
+    public void testCnn3DPlain() {
```

Comment:
IMO this test needs to be more robust:
1. Test at least some different input d/h/w - with different values for both. Why? In case there's some issue with native code (using wrong arg/value, which won't be picked up with all same values)
2. Ditto for testing multile kernels/strides - I'd do something like 1/1/1, 2/2/2, 1/2/3

Reply:
@AlexDBlack of course, I'll do that once this first sanity check works.